wdl-lint/src/rules/import_sort.rs

```diff
@@ -79,27 +82,67 @@ impl Visitor for ImportSortRule {
+        // Collect all import statements
+        let imports: Vec<_> = doc
             .syntax()
             .children_with_tokens()
-            .filter(|c| c.kind() == SyntaxKind::ImportStatementNode)
-            .map(|c| c.into_node().unwrap());
-
-        let mut prev_import: Option<SyntaxNode> = None;
-        for import in imports {
-            if let Some(prev) = prev_import {
-                if import.text().to_string() < prev.text().to_string() {
-                    // Since this rule can only be excepted in a document-wide fashion,
-                    // if the rule is running we can directly add the diagnostic
-                    // without checking for the exceptable nodes
-                    state.add(import_not_sorted(import.text_range().to_span()));
-                    return; // Only report one sorting diagnostic at a time.
-                }
-            }
-            prev_import = Some(import);
+            .filter(|n| n.kind() == SyntaxKind::ImportStatementNode)
+            .filter_map(|c| c.into_node())
+            .collect();
+
+        if imports.is_empty() {
+            return;
+        }
+
+        // Clone imports for comparison
+        let mut sorted_imports = imports.clone();
+        sorted_imports.sort_by(|a, b| {
+            let a_uri = ImportStatement::cast(a.clone())
+                .expect("import statement")
+                .uri()
+                .text()
+                .expect("import uri");
+            let b_uri = ImportStatement::cast(b.clone())
+                .expect("import statement")
+                .uri()
+                .text()
+                .expect("import uri");
+            a_uri.as_str().cmp(b_uri.as_str())
+        });
+
+        // Check if sorting is needed
+        if imports != sorted_imports {
+            // Find the first out-of-order import
+            let first_out_of_order = sorted_imports
+                .iter()
+                .find(|sorted_import| {
+                    let original_position = imports.iter().position(|orig| orig == *sorted_import);
+                    let sorted_position = sorted_imports
+                        .iter()
+                        .position(|sorted| sorted == *sorted_import);
+                    original_position != sorted_position
+                })
+                .unwrap();
+
+            let span = imports
+                .iter()
+                .find(|orig| **orig == *first_out_of_order)
+                .unwrap()
+                .text_range()
+                .to_span();
+            let sorted_imports_text = sorted_imports
+                .iter()
+                .map(|import| import.text().to_string())
+                .collect::<Vec<_>>()
+                .join("\n");
+
+            state.exceptable_add(
+                import_not_sorted(span, sorted_imports_text),
+                SyntaxElement::from(imports.first().unwrap().clone()),
+                &self.exceptable_nodes(),
+            );
```

Comment:
```suggestion
            let span = imports
                .iter()
                .zip(sorted_imports.iter())
                .find(|(a, b)| a != b)
                .expect("first out-of-order import")
                .0
                .first_token()
                .expect("node should have a first token")
                .text_range()
                .to_span();
            state.add(import_not_sorted(
                span,
                sorted_imports
                    .iter()
                    .map(|i| i.text().to_string())
                    .collect::<Vec<_>>()
                    .join("\n"),
            ));
```
what you have works and is fine, but I've modified it a bit to be a little more efficient.

Reply:
@a-frantz i tried running this but the errors seems to be diff, will need to change the test files for this 
`
Diff < left / right > :
 note[ImportSort]: imports are not sorted lexicographically
<  ┌─ tests/lints/within-import-whitespace/source.wdl:9:1
>  ┌─ tests/lints/within-import-whitespace/source.wdl:8:1
   │
<9 │ import  "bar"  # BAD (2 spaces)
<  │ ^^^^^^^^^^^^^
>8 │ import "foo" as foo  # OK
>  │ ^^^^^^
`

Reply:
Please commit my suggested change and then re-BLESS the tests locally, then push

Reply:
okay got it